api/filtering/db_filters.py

```diff
@@ -195,6 +196,12 @@ def _timestamp_and_host_type_filter(condition, state):
     return (_timestamp_and_host_type_filter(condition, state) for state in filtered_states)
 
 
+def find_stale_host_in_window(staleness, host_type, last_run_secs, timestamp_filter_func):
+    logger.debug("finding hosts that went stale in the last %s seconds", last_run_secs)
+    host_stale = HostStale(staleness, host_type, last_run_secs)
+    return (timestamp_filter_func(*host_stale.stale_in_window()),)
```

Comment:
IMO the HostStale class doesn't need to exist. We also don't need to pass in a `timestamp_filter_func` bc it's going to be the same every time. For example:
```suggestion
def find_stale_host_in_window(staleness, host_type, last_run_secs):
    logger.debug("finding hosts that went stale in the last %s seconds", last_run_secs)
    # TODO: We'll want to pass in a value for "end_date"
    end_date = datetime.now()
    prefix = "immutable" if host_type == "edge" else "conventional"
    stale_timestamp = end_date - timedelta(seconds=staleness[f"{prefix}_time_to_stale"])
    return stale_timestamp_filter(
        stale_timestamp - timedelta(seconds=last_run_secs),
        stale_timestamp,
    )
```